pkg/apiserver/filters/impersonation.go

```diff
@@ -123,6 +123,15 @@ func WithImpersonation(handler http.Handler, requestContextMapper api.RequestCon
 		oldUser, _ := api.UserFrom(ctx)
 		httplog.LogOf(req, w).Addf("%v is acting as %v", oldUser, newUser)
 
+		// clear all the impersonation headers from the request
+		req.Header.Del(authenticationapi.ImpersonateUserHeader)
```

Comment:
What about when it is a different header?
Ref https://github.com/kubernetes/kubernetes/pull/36816/ that adds a flag to customize the header.